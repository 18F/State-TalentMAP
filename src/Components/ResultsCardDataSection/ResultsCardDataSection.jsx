import React from 'react';
import { Link } from 'react-router-dom';
import { POSITION_DETAILS } from '../../Constants/PropTypes';
import * as SystemMessages from '../../Constants/SystemMessages';
import ResultsCardDataItem from '../ResultsCardDataItem/ResultsCardDataItem';
import { shortenString } from '../../utilities';

const ResultsCardDataSection = ({ result }) => (
  <div>
    <div className="usa-width-five-twelfths data-section-left-container">
      <div className="data-section-left">
        <ResultsCardDataItem
          title="Post Information"
          items={
          [
            { description: 'Post', text: result.post ? <Link to={`/post/${result.post.id}`}>{result.post.location}</Link> : SystemMessages.NO_POST },
            { description: 'Bureau', text: result.bureau },
            { description: 'Post Differential',
              text: result.post
                ? result.post.differential_rate : SystemMessages.NO_POST_DIFFERENTIAL },
          ]
          }
        />
      </div>
    </div>
    <div className="usa-width-five-twelfths data-section-right-container">
      <div className="data-section-right">
        <ResultsCardDataItem
          title="Position Information"
          items={
          [
            { description: 'Skill Code', text: result.skill },
            { description: 'Grade', text: result.grade },
            { description: 'Description',
              text: result.description && result.description.content ?
                shortenString(result.description.content, 35) :
                SystemMessages.NO_POSITION_DESCRIPTION },
          ]
          }
        />
      </div>
    </div>
  </div>
);

ResultsCardDataSection.propTypes = {
  result: POSITION_DETAILS.isRequired,
};

export default ResultsCardDataSection;
