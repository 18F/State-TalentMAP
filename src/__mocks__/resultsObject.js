const resultsObject = {
  count: 2,
  results: [
    { id: 6,
      grade: '05',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      description: {
        content: 'The quick brown fox jumped over the lazy dog.',
        date_created: '2018-01-12T15:09:22.952226Z',
      },
      post: { id: 162, obc_id: 162, tour_of_duty: '2YRR', code: 'LT6000000', location: 'MASERU, LESOTHO', cost_of_living_adjustment: 0, differential_rate: 15, danger_pay: 0, rest_relaxation_point: 'London', has_consumable_allowance: false, has_service_needs_differential: false },
      languages: [
        { id: 1, language: 'French (FR)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'French (FR) 2/2' },
        { id: 1, language: 'German (GM)', reading_proficiency: '2', spoken_proficiency: '2', representation: 'German (GM) 2/2' },
      ],
      title: 'Position - First',
    },
    { id: 60,
      grade: '03',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '150000',
      organization: 'YAOUNDE CAMEROON (YAOUNDE)',
      position_number: '00025003',
      is_overseas: true,
      create_date: '2006-09-20',
      update_date: '2017-06-08',
      bid_statistics: [
        {
          id: 1,
          total_bids: 3,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
          bidcycle: 'Demo BidCycle 2018-01-12 15:11:38.004793',
        },
      ],
      post: null,
      languages: [],
      title: 'Position - Second',
    },
  ],
};

export default resultsObject;
