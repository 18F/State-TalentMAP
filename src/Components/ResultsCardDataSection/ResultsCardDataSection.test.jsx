import { shallow } from 'enzyme';
import React from 'react';
import TestUtils from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import ResultsCardDataSection from './ResultsCardDataSection';
import resultsObject from '../../__mocks__/resultsObject';

describe('ResultsCardDataSectionComponent', () => {
  it('is defined', () => {
    const wrapper = TestUtils.renderIntoDocument(<MemoryRouter>
      <ResultsCardDataSection result={resultsObject.results[0]} />
    </MemoryRouter>);
    expect(wrapper).toBeDefined();
  });

  it('can receive props', () => {
    const wrapper =
      shallow(<ResultsCardDataSection result={resultsObject.results[0]} />);
    expect(wrapper.instance().props.result.id).toBe(6);
  });

  it('can receive different types of results', () => {
    const wrapper =
      shallow(<ResultsCardDataSection result={resultsObject.results[1]} />);
    expect(wrapper.instance().props.result.id).toBe(60);
  });

  it('matches snapshot', () => {
    const wrapper =
      shallow(<ResultsCardDataSection result={resultsObject.results[1]} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
