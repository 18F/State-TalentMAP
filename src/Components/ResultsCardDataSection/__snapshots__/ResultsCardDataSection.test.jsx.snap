// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCardDataSectionComponent matches snapshot 1`] = `
<div>
  <div
    className="usa-width-five-twelfths data-section-left-container"
  >
    <div
      className="data-section-left"
    >
      <ResultsCardDataItem
        items={
          Array [
            Object {
              "description": "Post",
              "text": "None listed",
            },
            Object {
              "description": "Bureau",
              "text": "150000",
            },
            Object {
              "description": "Post Differential",
              "text": "None listed",
            },
          ]
        }
        title="Post Information"
      />
    </div>
  </div>
  <div
    className="usa-width-five-twelfths data-section-right-container"
  >
    <div
      className="data-section-right"
    >
      <ResultsCardDataItem
        items={
          Array [
            Object {
              "description": "Skill",
              "text": "OFFICE MANAGEMENT (9017)",
            },
            Object {
              "description": "Grade",
              "text": "03",
            },
            Object {
              "description": "Description",
              "text": "There is no description for this position.",
            },
          ]
        }
        title="Position Information"
      />
    </div>
  </div>
</div>
`;
